### DISCUSSION TITLE
Can't create a tunnel termination via the API

### DISCUSSION THREAD

**Discussion by Edward033** [netbox-community/netbox#17002] [Q&A]
Hi all! 

Anyone recently created a VPN tunnel termination via the API? Unsure if I'm running into a bug, or potentially missing a parameter: 

curl -s -X POST -H "Authorization: Token $TOKEN" -H "Content-Type: application/json" http://netbox.com/api/vpn/tunnel-terminations/ --data '{"tunnel":1, "termination_type":"dcim.interface", "role":"peer","outside_ip": 15100, "tags": [1]}'

{"error": "'NoneType' object has no attribute '_meta'", "exception": "AttributeError", "netbox_version": "3.7.0", "python_version": "3.8.17"}%

I've created tunnels without any problems, but stump with tunnel terminations. 

Thanks!

**Comment by candlerb** [comment]
What do you see if you GET an existing tunnel termination?

However, before looking much further I think you should first upgrade to the latest 3.7 (i.e. 3.7.8) to see if it has been fixed. You are using a new feature which was only introduced in 3.7.0, and the release notes suggest quite a lot was fixed since then:

```
% grep tunnel docs/release-notes/version-3.7.md
* [#15654](https://github.com/netbox-community/netbox/issues/15654) - Fix `AttributeError` exception when attempting to save an incomplete tunnel termination
* [#14707](https://github.com/netbox-community/netbox/issues/14707) - Clarify interface designation when creating tunnel terminations
* [#15291](https://github.com/netbox-community/netbox/issues/15291) - Add tunnel termination buttons to VM interfaces table
* [#15126](https://github.com/netbox-community/netbox/issues/15126) - `group` field should be optional when creating VPN tunnel via REST API
* [#15127](https://github.com/netbox-community/netbox/issues/15127) - Add missing group column to VPN tunnels table
* [#14839](https://github.com/netbox-community/netbox/issues/14839) - Fix form validation error when attempting to terminate a tunnel to a virtual machine interface
* [#14885](https://github.com/netbox-community/netbox/issues/14885) - Add missing "group" field to VPN tunnel creation form
* [#14663](https://github.com/netbox-community/netbox/issues/14663) - Fix tunnel creation when setting initial termination to a VM interface
* [#14706](https://github.com/netbox-community/netbox/issues/14706) - Relax one-to-one mapping of tunnel termination to IP address
* [#14709](https://github.com/netbox-community/netbox/issues/14709) - Fix typo in tunnel termination type choice name
Several new models have been introduced to enable [VPN tunnel management](../features/vpn-tunnels.md). Users can now define tunnels with two or more terminations to represent peer-to-peer or hub-and-spoke topologies. Each termination is made to a virtual interface on a device or virtual machine. Additionally, users can define IKE and IPSec proposals and policies, which can be applied to tunnels to document encryption and authentication strategies.
    * `/api/vpn/tunnels/`
    * `/api/vpn/tunnel-terminations/`
```

Upgrading between Netbox patch releases, e.g. 3.7.x to 3.7.y, is generally a low-risk upgrade with mostly bug fixes and minor feature improvements, although historically there have been a few exceptions to that rule.

The rule that the version ending .0 of anything is probably buggy works well for most things though :-)
